community/lucene-index/src/main/java/org/neo4j/index/impl/lucene/legacy/IndexType.java

```diff
@@ -374,6 +374,7 @@ final void removeFromDocument( Document document, String key, Object value )
         else
         {
             removeFieldsFromDocument( document, key, value );
+            restoreSortFields( document );
```

Comment:
Is this a fix for something unrelated to the original PR changes?

Reply:
Not unrelated. This is currently the only _known_ cause for the case handled in the first commit. One of the tests in this PR tests for this bug explicitly.